api/src/main/java/com/basistech/rosette/RosetteFactoryConfigurationProperty.java

```diff
@@ -0,0 +1,40 @@
+/*
+* Copyright 2017 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.basistech.rosette;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD})
+/**
+ * This annotation marks factory configuration options that should be included in configuration
+ * files in an on-premise deployment.
+ */
+public @interface RosetteFactoryConfigurationProperty {
+    /**
+     * The default value of the configuration option as a yaml string.
+     */
+    String defaultValue() default "";
```

Comment:
I'm not thinking about customers, I'm thinking about a SDK developer. If my factory has an option of `RosetteComplexOption` type, ideally I'd want to do something like `@RosetteFactoryConfigurationProperty(default=RosetteComplexOption.DEFAULT_VALUE)` any hope the annotation magic can serialize or toString it for me. But I guess this may not be achievable without a bunch of work.